### PR TITLE
add koonovo 3D printers

### PR DESCRIPTION
**Pull request by KOONOVO3DPrinter**

### CODE REVIEW

File: resources/definitions/KOONOVOELF.def.json [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    "version": 2,
+    "name": "KOONOVO ELF",
+    "inherits": "fdmprinter",
+    "metadata": {
+        "visible": true,
+        "author": "KOONOVO",
+        "manufacturer": "KOONOVO",
+        "weight": 4,
+        "file_formats": "text/x-gcode",
+        "icon": "icon_ultimaker.png",
```
Comment (fieldOfView) [inline resources/definitions/KOONOVOELF.def.json:11]: This line should be removed

File: resources/definitions/KOONOVOELF.def.json [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+        "platform": "KOONOVO.obj",
+        "platform_texture": "KOONOVO.png",
+        "has_materials": true,
+        "preferred_material": "generic_pla",
+        "has_machine_quality": true,
+        "quality_definition": "KOONOVO_ELF",
```
Comment (fieldOfView) [inline resources/definitions/KOONOVOELF.def.json:18]: KOONOVO_ELF is your extruder definition. You cannot refer to the extruder definition for qualities.

File: resources/definitions/KOONOVOELF.def.json [hunk 3]
```diff
@@ -0,0 +1,94 @@
+{
+    "version": 2,
+    "name": "KOONOVO ELF",
+    "inherits": "fdmprinter",
+    "metadata": {
+        "visible": true,
+        "author": "KOONOVO",
+        "manufacturer": "KOONOVO",
+        "weight": 4,
```
Comment (fieldOfView) [inline resources/definitions/KOONOVOELF.def.json:9]: This line should be removed

File: resources/definitions/KOONOVOELF.def.json [hunk 4]
```diff
@@ -0,0 +1,94 @@
+{
+    "version": 2,
+    "name": "KOONOVO ELF",
+    "inherits": "fdmprinter",
+    "metadata": {
+        "visible": true,
```
Comment (fieldOfView) [inline resources/definitions/KOONOVOELF.def.json:6]: This line is unnecessary

File: resources/definitions/KOONOVOELF.def.json [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+        {
+            "default_value": [
+                [ -75, 35 ],
+                [ -75, -18 ],
+                [ 18, 35 ],
+                [ 18, -18 ]
+            ]
+        },
+        "gantry_height": {
+            "default_value": 55
```
Comment (fieldOfView) [inline resources/definitions/KOONOVOELF.def.json:70]: Use `"value": "55"` instead

File: resources/quality/KOONOVO_ELF_draft.inst.cfg [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+
+[metadata]
+setting_version = 16
+type = quality
+quality_type = draft
+weight = -2
+global_quality = True
+
+[values]
+acceleration_enabled = True
```
Comment (nallath) [inline resources/quality/KOONOVO_ELF_draft.inst.cfg:14]: These values are the same everywhere. It's better to just leave them in your printer definition.

File: resources/definitions/KOONOVOPyramid.def.json [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+    "version": 2,
+    "name": "KOONOVO Pyramid",
+    "inherits": "fdmprinter",
+    "metadata": {
+        "visible": true,
+        "author": "KOONOVO",
+        "manufacturer": "KOONOVO",
+        "weight": 4,
+        "file_formats": "text/x-gcode",
+        "icon": "icon_ultimaker.png",
```
Comment (fieldOfView) [inline resources/definitions/KOONOVOPyramid.def.json:11]: Remove this line

File: resources/extruders/KOONOVO_ELF.def.json [hunk 8]
```diff
@@ -0,0 +1,28 @@
+{
+    "id": "KOONOVO_ELF",
+    "version": 2,
+    "name": "Extruder 1",
+    "inherits": "fdmextruder",
+    "metadata": {
+        "machine": "KOONOVO",
```
Comment (fieldOfView) [inline resources/extruders/KOONOVO_ELF.def.json:7]: This refers to a .def.json that does not exist. I think you mean KOONOVOELF

**Comment by KOONOVO3DPrinter** [comment]
&nbsp;delete&nbsp; &nbsp;"icon": "icon_ultimaker.png", is ok
&nbsp;
------------------&nbsp;Original&nbsp;------------------

**Comment by KOONOVO3DPrinter** [comment]
&nbsp;delete&nbsp;weight ?
&nbsp;
&nbsp;
------------------&nbsp;Original&nbsp;------------------
From: "Aldo Hoeben"; 
Date: 2020年9月22日(星期二) 下午3:50
To: "Ultimaker/Cura"; 
Cc: "KOONOVO3DPrinter"; "Author"; 
Subject: Re: [Ultimaker/Cura] add koonovo 3D printers (#8425)

&nbsp;


 
@fieldOfView commented on this pull request.
 
 
In resources/definitions/KOONOVOELF.def.json:
 &gt; +    "name": "KOONOVO ELF", +    "inherits": "fdmprinter", +    "metadata": { +        "visible": true, +        "author": "KOONOVO", +        "manufacturer": "KOONOVO", +        "weight": 4, +        "file_formats": "text/x-gcode", +        "icon": "icon_ultimaker.png", + +        "platform": "KOONOVO.obj", +        "platform_texture": "KOONOVO.png", +        "has_materials": true, +        "preferred_material": "generic_pla", +        "has_machine_quality": true, +        "quality_definition": "KOONOVO_ELF",  
KOONOVO_ELF is your extruder definition. You cannot refer to the extruder definition for qualities.

File: resources/definitions/KOONOVOELF.def.json (hunk 1 above)
Comment (KOONOVO3DPrinter) [reply resources/definitions/KOONOVOELF.def.json]: > 该行应删除

delete “ icon ”： “ icon_ultimaker.png ”，

File: resources/definitions/KOONOVOELF.def.json (hunk 3 above)
Comment (KOONOVO3DPrinter) [reply resources/definitions/KOONOVOELF.def.json]: > 该行应删除

delete weihgt

File: resources/definitions/KOONOVOELF.def.json (hunk 5 above)
Comment (KOONOVO3DPrinter) [reply resources/definitions/KOONOVOELF.def.json]: > 使用`"value": "55"`替代
“ default_value ”：value

File: resources/quality/KOONOVO_ELF_draft.inst.cfg (hunk 6 above)
Comment (KOONOVO3DPrinter) [reply resources/quality/KOONOVO_ELF_draft.inst.cfg:14]: > These values are the same everywhere. It's better to just leave them in your printer definition.
definition in KOONOVOELF.def.json

File: resources/definitions/KOONOVOPyramid.def.json (hunk 7 above)
Comment (KOONOVO3DPrinter) [reply resources/definitions/KOONOVOPyramid.def.json]: > Remove this line

delete "icon": "icon_ultimaker.png",

File: resources/extruders/KOONOVO_ELF.def.json (hunk 8 above)
Comment (KOONOVO3DPrinter) [reply resources/extruders/KOONOVO_ELF.def.json]: "machine": "KOONOVOELF",

**Comment by fieldOfView** [comment]
There's a convention to use lowercase filenames instead of uppercase. So `koonovo_elf.def.json` instead of `KOONOVOELF.def.json`. I would also suggest you use the `_extruder_0` convention in the filename of the extruder.

**Comment by KOONOVO3DPrinter** [comment]
OK，Thanks



&nbsp;
&nbsp;
------------------&nbsp;Original&nbsp;------------------
From: "Aldo Hoeben"; 
Date: 2020年9月22日(星期二) 下午4:29
To: "Ultimaker/Cura"; 
Cc: "KOONOVO3DPrinter"; "Author"; 
Subject: Re: [Ultimaker/Cura] add koonovo 3D printers (#8425)

&nbsp;


 
There's a convention to use lowercase filenames instead of uppercase. So koonovo_elf.def.json instead of KOONOVOELF.def.json. I would also suggest you use the _extruder_0 convention in the filename of the extruder.

File: resources/definitions/koonovo_elf.def.json [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+        {
+            "default_value": [
+                [ -75, 35 ],
+                [ -75, -18 ],
+                [ 18, 35 ],
+                [ 18, -18 ]
+            ]
+        },
+        "gantry_height": {
+            "default_value": 0
```
Comment (konskarm) [inline resources/definitions/koonovo_elf.def.json:68]:
```suggestion
            "value": 0
```

File: resources/definitions/koonovo_pyramid.def.json [hunk 10]
```diff
@@ -0,0 +1,92 @@
...
+        {
+            "default_value": [
+                [ -75, 35 ],
+                [ -75, -18 ],
+                [ 18, 35 ],
+                [ 18, -18 ]
+            ]
+        },
+        "gantry_height": {
+            "default_value": 0
```
Comment (konskarm) [inline resources/definitions/koonovo_pyramid.def.json:68]:
```suggestion
            "value": 0
```

File: resources/extruders/koonovo_elf.def.json [hunk 11]
```diff
@@ -0,0 +1,28 @@
+{
+    "id": "koonovo_elf",
```
Comment (konskarm) [inline resources/extruders/koonovo_elf.def.json:2]:
```suggestion
```

**Review by konskarm — COMMENTED** [review]
Issues reported by the test system that need to be fixed.

**Pull request closed without merging**